File: examples/typescript/data/initial/device.ts
import { factory } from '../../model/factory';
import { Device, Android, Ios } from '../../model/model';
import { DATAID } from '../dataid';

export function registerInitialDevice() {
  const f = factory;
  f.register(DATAID.TEST01_DEVICE_1, () => {
    return {
      android: f.childNode(DATAID.TEST01_ANDROID_1) as Android,
      ios: f.childNode(DATAID.TEST01_IOS_1) as Ios,
    } as Device;
  });
  f.register(DATAID.TEST05_DEVICE_1, () => {
    return {} as Device;
  });
}

File: examples/golang/data/initial/device.go
// +build test

package initial

import (
	"forcedata/examples/golang/model"
)

// データの識別子
const (
	Test01Device1 model.DataID = "Test01Device1"
	Test05Device1 model.DataID = "Test05Device1"
)

func RegisterDevice() {
	f := model.Factory
	f.Register(Test01Device1, func() interface{} {
		return &model.Device{
			Android: f.ChildNode(Test01Android1).(*model.Android),
			Ios:     f.ChildNode(Test01Ios1).(*model.Ios),
		}
	})
	f.Register(Test05Device1, func() interface{} {
		return &model.Device{}
	})
}
